### PR TITLE
Move mod/wall_upload.php to src/Module

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #4090

Note: This POST-only module is supposed to return `{"ok": true}` when the upload is successful and the `response=json` parameter is provided, but I was only able to get this behavior once in all my tests, never to happen again. Otherwise, the response body is always `<html></html>` which isn't correct. However, this unexpected behavior also happens in the current `mod/wall_upload.php` module, so I didn't dig deeper (and I dug pretty deep into the ApiResponse/Guzzle\Psr7\Response stack).

### CODE REVIEW

File: src/Module/Profile/Photos/Upload.php [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+			$can_post   = $this->database->exists('contact', ['blocked' => false, 'pending' => false, 'id' => $contact_id, 'uid' => $page_owner_uid]);
+			$visitor    = $contact_id;
+		}
+
+		if (!$can_post) {
+			$this->logger->warning('No permission to upload files', ['contact_id' => $contact_id, 'page_owner_uid' => $page_owner_uid]);
+			$this->exit(403, $this->l10n->t('Permission denied.'), true);
+		}
+
+		if (empty($_FILES['userfile']) && empty($_FILES['media'])) {
```
Comment (nupplaphil) [inline src/Module/Profile/Photos/Upload.php:111]: There's a specific class for files, @annando has created under Utils, I'm currently on my mobile phone so I cannot exactly say which one it is :D

File: src/Module/Profile/Photos/Upload.php [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+		if (!strlen($album)) {
+			$album = $this->l10n->t('Wall Photos');
+		}
+
+		$defperm = '<' . $default_cid . '>';
+
+		$result = Photo::store($image, $page_owner_uid, $visitor, $resource_id, $filename, $album, 0, Photo::DEFAULT, $defperm);
+		if (!$result) {
+			$this->logger->warning('Photo::store() failed', ['result' => $result]);
+			$this->exit(401, $this->l10n->t('Image upload failed.'));
```
Comment (nupplaphil) [inline src/Module/Profile/Photos/Upload.php]:
```suggestion
			$this->exit(401, $this->t('Image upload failed.'));
```

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Module/Profile/Photos/Upload.php [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+	}
+
+	/**
+	 * @param int    $httpCode
+	 * @param string $message
+	 * @param bool   $systemMessage
+	 * @return void
+	 * @throws InternalServerErrorException
+	 */
+	private function exit(int $httpCode, string $message, bool $systemMessage = false): void
```
Comment (nupplaphil) [inline src/Module/Profile/Photos/Upload.php]: Can you check if `$this->response` is useful here?

As discussed yesterday, it's currently not fully useful, but in the end, everything should be a Response and no module or class should directly exit the process -> testability:-)

**Review by nupplaphil — CHANGES_REQUESTED** [review]

File: src/Module/Profile/Photos/Upload.php (hunk 2 above)
Comment (MrPetovan) [reply src/Module/Profile/Photos/Upload.php]: Right, I always forget there are BaseModule wrappers for these.

File: src/Module/Profile/Photos/Upload.php (hunk 3 above)
Comment (MrPetovan) [reply src/Module/Profile/Photos/Upload.php]: After reviewing what happens with the embedded `$this->response` in `BaseModule->run()`, I ended up simplifying the exit method and removing the dependency to the `Core\System` class. 👍

File: src/Module/Profile/Photos/Upload.php (hunk 1 above)
Comment (MrPetovan) [reply src/Module/Profile/Photos/Upload.php:111]: I could only find `Util\HttpInput` but it looks like the files array are just added to the `$request` array provided to `post()`, although I wasn't able to confirm it.

**Comment by nupplaphil** [comment]
What happened @MrPetovan ?

**Comment by MrPetovan** [comment]
I renamed the branch because of the double `move` in the name 😅

**Pull request closed without merging**